### ISSUE TITLE
Expose AppDelegate callbacks for universal modules

### ISSUE THREAD

**Issue by sjchmiela** [expo/expo#2706]
vide https://github.com/expo/expo/pull/2543/files#r231172050

**Comment by EvanBacon** [comment]
Also for notifications https://github.com/expo/expo/tree/master/packages/expo-firebase-notifications#common-setup
```js
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(nonnull NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler;
```

and Linking: https://github.com/expo/expo/tree/master/packages/expo-firebase-invites#common-setup
```js
- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url options:(NSDictionary<NSString *, id> *)options;
- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *))restorationHandler;
```

**Comment by tsapeta** [comment]
Closing - it's been fixed by #3917.
